gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/GemFireCacheImpl.java

```diff
@@ -605,7 +605,7 @@ public void addOldEntry(RegionEntry oldRe, RegionEntry newEntry, LocalRegion reg
     if (!region.reservedTable() && region.needAccounting()) {
       region.calculateEntryOverhead(oldRe);
       LocalRegion.regionPath.set(region.getFullPath());
-      region.acquirePoolMemory(0, oldSize, forDelete, null, true);
+      region.acquirePoolMemory(0, oldRe.getValueSize(), oldRe.isForDelete(), null, true);
```

Comment:
@rishitesh If that's what you want to account then need to add the overhead of NonLocalRegionEntry. However, the update call to acquirePool was skipping overhead completely (the "forDelete" flag) while free was adding. Perhaps can add NonLocalRegionEntry overhead fo oldRe.valueSize before acquire call then it will get accounted in both places now.